src/pages/home/report/ReportActionItemMessage.js

```diff
@@ -47,25 +48,42 @@ function ReportActionItemMessage(props) {
         }
     }
 
+    const isApprovedOrSubmittedReportActionType = _.contains([CONST.REPORT.ACTIONS.TYPE.APPROVED, CONST.REPORT.ACTIONS.TYPE.SUBMITTED], props.action.actionName);
+
+    const flaggedContentText = <Text style={[styles.textLabelSupporting, styles.lh20]}>{props.translate('moderation.flaggedContent')}</Text>;
+
+    /**
+     * Get a ReportActionItemFragment
+     * @param {Object} fragment the current message fragment
+     * @param {Number} index the current message fragment's index
+     * @returns {Object} report action item fragment
+     */
+    const renderReportActionItemFragment = (fragment, index) => (
+        <ReportActionItemFragment
+            key={`actionFragment-${props.action.reportActionID}-${index}`}
+            fragment={fragment}
+            iouMessage={iouMessage}
+            isThreadParentMessage={ReportActionsUtils.isThreadParentMessage(props.action, props.reportID)}
+            attachmentInfo={props.action.attachmentInfo}
+            pendingAction={props.action.pendingAction}
+            source={lodashGet(props.action, 'originalMessage.source')}
+            accountID={props.action.actorAccountID}
+            style={props.style}
+            displayAsGroup={props.displayAsGroup}
+            actionName={props.action.actionName}
+        />
+    );
+
     return (
         <View style={[styles.chatItemMessage, !props.displayAsGroup && isAttachment ? styles.mt2 : {}, ...props.style]}>
-            {!props.isHidden ? (
-                _.map(messages, (fragment, index) => (
-                    <ReportActionItemFragment
-                        key={`actionFragment-${props.action.reportActionID}-${index}`}
-                        fragment={fragment}
-                        iouMessage={iouMessage}
-                        isThreadParentMessage={ReportActionsUtils.isThreadParentMessage(props.action, props.reportID)}
-                        attachmentInfo={props.action.attachmentInfo}
-                        pendingAction={props.action.pendingAction}
-                        source={lodashGet(props.action, 'originalMessage.source')}
-                        accountID={props.action.actorAccountID}
-                        style={props.style}
-                        displayAsGroup={props.displayAsGroup}
-                    />
-                ))
+            {isApprovedOrSubmittedReportActionType ? (
+                // Wrapping 'ReportActionItemFragment' inside '<Text>' so that text isn't broken up into separate lines when
+                // there are multiple messages of type 'TEXT', as seen when a report is submitted/approved from a
+                // policy on Old Dot and then viewed on New Dot.
```

Comment:
```suggestion
                // Approving or submitting reports in oldDot results in system messages made up of multiple fragments of `TEXT` type
                // which we need to wrap in `<Text>` to prevent them rendering on separate lines.
```